### PR TITLE
[Tablet Orders] Update scan button rendering based on size class

### LINKED ISSUE #11747: [Tablet Orders] Hide `+` and `scan` buttons for add product on the Order Summary when shown in a side-by-side view.
Figma: 6kkCwI9VOEYcidp6UT5bln-fi-192_25891

Screenshot 2024-01-22 at 15 39 49

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #11747

## Description
This PR addresses how we render the "scan" button from Order creation based on size class: 

The existing design shows a scanner button in the Order Summary screen, either in the "Add Products" row when no products have been added to the order yet, or in the component that holds products:
| Existing design: Order has no products | Existing design: Order has products |
|--------|--------|
| Screenshot 2024-02-15 at 14 58 54 | Screenshot 2024-02-15 at 14 59 07 | 

The proposed design for tablets changes this by removing the scan button from the "Order Summary" view, and adding it to the "Product Selector" view instead, to the right side of the search bar:
| Proposed changes |
|--------|
| Screenshot 2024-02-15 at 14 58 02 |

## Testing instructions
* Keep the `sideBySideViewForOrderForm` feature flag set to `false`
* On an iPhone that does not support split views (eg: iPhone 15), go to Orders > Tap `+` > Observe that the scanner button appears as always, both when no products are added to the order, as well as when products are added. Confirm both in portrait and landscape mode.
* Enable the `sideBySideViewForOrderForm` feature flag set to `true`
* Repeat the previous step. You should see the same results.
* On a tablet, go to Orders > Tap `+` > Observe that the scanner shows in the product selector, near to search products, and does not appear in the order summary.
* Tapping the button will not trigger the scanning flow, this will be tackled separately on #12011

## Caveats
* As mentioned in the testing instructions, tapping the button will not trigger the scanning flow, this will be tackled separately on #12011.
* The polished design of the header has not been tackled yet, this will be done on #11748, this PR only deals with hiding and showing the scan button when appropriate based on size class.

_Side-note: I found useful to test as well on an iPhone that supports split views to see if the render works correctly (like the 15 Plus) while the design is not the best yet due to the small screen and there are some bugs, the portrait uses ModalOnModal while landscape uses SideBySide._

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift [hunk 1]
```diff
@@ -554,7 +559,7 @@ private struct ProductsSection: View {
 
                     HStack(spacing: OrderForm.Layout.productsHeaderButtonsSpacing) {
                         scanProductButton
-                        .renderedIf(viewModel.isAddProductToOrderViaSKUScannerEnabled)
+                            .renderedIf(viewModel.isAddProductToOrderViaSKUScannerEnabled && presentationStyle == .modalOnModal)
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift]: I'm planning to remove the check for `viewModel.isAddProductToOrderViaSKUScannerEnabled` next, this feature has been live for quite a while now and having the feature flag check here and through other views increases its complexity for no benefit.

File: WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift [hunk 2]
```diff
@@ -83,20 +83,30 @@ struct ProductSelectorView: View {
...
-                        .renderedIf(searchHeaderisBeingEdited)
+                Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
+                    ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+                }
+                .pickerStyle(.segmented)
+                .padding(.leading)
+                .padding(.trailing)
+                .renderedIf(searchHeaderisBeingEdited)
+                Button(action: {
+                    // TODO: gh-12011
```
Comment (bozidarsevo) [inline WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift:100]: is this left to be implemented here or?

Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift:100]: Sorry, I should have written the full link 😅  Yes, will be implemented separately on issue 12011: https://github.com/woocommerce/woocommerce-ios/issues/12011

**Review by bozidarsevo — APPROVED** [review]

**Pull request merged**